src/query/sql/src/planner/binder/udf.rs

```diff
@@ -0,0 +1,106 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_ast::ast::AlterUDFStmt;
+use common_ast::ast::CreateUDFStmt;
+use common_ast::ast::Identifier;
+use common_ast::ast::UDFDefinition;
+use common_exception::Result;
+use common_expression::types::DataType;
+use common_meta_app::principal::LambdaUDF;
+use common_meta_app::principal::UDFDefinition as PlanUDFDefinition;
+use common_meta_app::principal::UDFServer;
+use common_meta_app::principal::UserDefinedFunction;
+
+use crate::planner::resolve_type_name;
+use crate::planner::udf_validator::UDFValidator;
+use crate::plans::AlterUDFPlan;
+use crate::plans::CreateUDFPlan;
+use crate::plans::Plan;
+use crate::Binder;
+
+impl Binder {
+    pub(in crate::planner::binder) async fn bind_udf_definition(
+        &mut self,
+        udf_name: &Identifier,
+        udf_description: &Option<String>,
+        udf_definition: &UDFDefinition,
+    ) -> Result<UserDefinedFunction> {
+        match udf_definition {
+            UDFDefinition::LambdaUDF {
+                parameters,
+                definition,
+            } => {
+                let mut validator = UDFValidator {
+                    name: udf_name.to_string(),
+                    parameters: parameters.iter().map(|v| v.to_string()).collect(),
+                    ..Default::default()
+                };
+                validator.verify_definition_expr(definition)?;
+                Ok(UserDefinedFunction {
+                    name: validator.name,
+                    description: udf_description.clone().unwrap_or_default(),
+                    definition: PlanUDFDefinition::LambdaUDF(LambdaUDF {
+                        parameters: validator.parameters,
+                        definition: definition.to_string(),
+                    }),
+                })
+            }
+            UDFDefinition::UDFServer {
+                arg_types,
+                return_type,
+                address,
+            } => {
+                // TODO(ccl): check function arguments type and return type
+                let mut arg_datatypes = Vec::with_capacity(arg_types.len());
+                for arg_type in arg_types {
+                    arg_datatypes.push(DataType::from(&resolve_type_name(arg_type)?));
+                }
+
+                Ok(UserDefinedFunction {
+                    name: udf_name.to_string(),
+                    description: udf_description.clone().unwrap_or_default(),
+                    definition: PlanUDFDefinition::UDFServer(UDFServer {
+                        address: address.clone(),
```

Comment:
Is it necessary to check that address is a valid HTTP address?

Reply:
Yes, I will do it in subsequent pr.